ear-production-suite-plugins/lib/src/programme_store_adm_serializer.cpp

```diff
@@ -411,81 +394,27 @@ void ProgrammeStoreAdmSerializer::createTopLevelObject(
-        // find the one that references the correct channel format
-        auto allTrackFormats = doc->getElements<adm::AudioTrackFormat>();
-
-        // For each channel format create a AudioTrackUid that references the pack
-        // format. The audio object also needs to reference the channel format.
-        for(int i(0); i < channelFormats.size(); ++i) {
-          auto audioTrackUid = adm::AudioTrackUid::create();
-          hoaAudioObject->addReference(audioTrackUid);
-          audioTrackUid->setReference(packFormat);
-
-          // We need the channel format from our document,
-          // not the document admCommonDefinitionHelper is using!
-          auto channelFormatId = adm::AudioChannelFormatId(
-            adm::TypeDefinition::HOA,
-            adm::AudioChannelFormatIdValue(channelFormats[i]->id));
-          auto channelFormat = doc->lookup(channelFormatId);
-          assert(channelFormat);
-
-          audioTrackUid->setReference(channelFormat);
-          pluginMap.push_back({ metadata.input_instance_id(), metadata.routing() + i, hoaAudioObject, audioTrackUid });
-        }
-        serializedObjects[connectionId] = hoaAudioObject;
-      }
+    } 
+    else if (metadata.has_hoa_metadata() || metadata.has_ds_metadata()) {
+      int pfIdVal = metadata.has_ds_metadata()
+                        ? metadata.ds_metadata().packformatidvalue()
+                        : metadata.hoa_metadata().packformatidvalue();
 
-    } else if (metadata.has_ds_metadata()) {
-      auto layoutIndex = metadata.ds_metadata().layout();
-      if (layoutIndex >= 0 && layoutIndex < SPEAKER_SETUPS.size()) {
-        std::vector<std::string> speakerLabels;
-        std::vector<adm::AudioChannelFormatId> channelFormatIds;
-        auto& speakerSetup = SPEAKER_SETUPS.at(layoutIndex);
-
-        for (auto& speaker : speakerSetup.speakers) {
-          speakerLabels.push_back(speaker.label);
-          auto channelFormatId = speaker.channelFormatId;
-          channelFormatIds.push_back(adm::parseAudioChannelFormatId(channelFormatId));
-        }
-        auto objectHolder = add2076v2TailoredCommonDefinitionsObjectTo(
-            doc, metadata.name(),
-            adm::parseAudioPackFormatId(speakerSetup.packFormatId),
-            channelFormatIds, speakerLabels);
-
-        setInteractivity(*objectHolder.audioObject, object);
-        setImportance(*objectHolder.audioObject, object);
-        content.addReference(objectHolder.audioObject);
-        for (int i(0); i < objectHolder.audioTrackUids.size(); ++i) {
-          auto speakerLabel = speakerLabels.at(i);
-          pluginMap.push_back({ metadata.input_instance_id(), metadata.routing() + i, objectHolder.audioObject, objectHolder.audioTrackUids.at(speakerLabel) });
-        }
-        serializedObjects[connectionId] = objectHolder.audioObject;
+      auto presets = AdmPresetDefinitionsHelper::getSingleton();
+      auto objectHolder = presets->addPresetDefinitionObjectTo(
```

Comment:
Bug here - generating a pack format with type def fixed as 1 will work for DS, but this condition also covers HOA